pkg/sql/stmtdiagnostics/statement_diagnostics.go

```diff
@@ -128,6 +128,12 @@ func (r *Request) isConditional() bool {
 	return r.minExecutionLatency != 0
 }
 
+// continueCollecting returns true if we want to continue collecting bundles for
+// this request. Notably it doesn't check whether the request has expired.
+func (r *Request) continueCollecting(st *cluster.Settings) bool {
+	return collectUntilExpiration.Get(&st.SV) && r.samplingProbability != 0 && !r.expiresAt.IsZero()
```

Comment:
I'm to blame for this unfortunate leakage. Is there a way to assert closer to where samplingProbability is set that expiresAt is non-zero? (Or is that already done?)) That way this validation code doesn't have to leak elsewhere.